bootstrap4/.js/src/main/scala/io/udash/bootstrap/datepicker/UdashDatePicker.scala

```diff
@@ -37,6 +38,24 @@ final class UdashDatePicker private[datepicker](
   ).render
   private val jQInput = jQ(inp).asInstanceOf[UdashDatePickerJQuery]
 
+  locally {
+    // When a date picker gets appended to a DOM, its options and listeners should be initialized once again. Thus, all
+    // nodes with mutated children are examined whether there is a date picker defined within a corresponding DOM
+    // subtree. It is also substantially possible for any date picker to be reported by more than one `MutationRecord`
+    // since the `subtree` switch of `MutationObserverInit` is on, thus only the first update is considered (note
+    // `findOpt`) to avoid double initialization.
+    new MutationObserver((records, _) =>
+      records
+        .iterator
+        .flatMap(record => for {i <- 0 until record.addedNodes.length} yield record.addedNodes(i))
+        .findOpt {
+          case element: Element => element.querySelector(s"#$componentId") != null
+          case _ => false
+        }
+        .foreach(_ => setupDatePicker())
+    ).observe(document.body, MutationObserverInit(childList = true, subtree = true))
```

Comment:
Unfortunately we are not able to narrow down `target` because we need to setup a mutation observer **before** appending nodes to be watched to DOM. Since we operate on a generic component here, existence of any parent node cannot be assumed except for the "static" ones (such as `html` and `body`).

I think single mutation observer with `querySelector` is fair enough - such a solution is recommended by many experienced frontend devs.

Reply:
How about a "hacky" (yet less involving on our code side) solution for node adding/removal detection like: https://davidwalsh.name/detect-node-insertion ?

Reply:
Mutation Observer approach left because of [better benchmark results](https://output.jsbin.com/hibeji).